llvm/lib/Target/AIE/AIELiveRegs.cpp

```diff
@@ -0,0 +1,97 @@
+//===- AIELiveRegs.cpp - Liveness Analysis infrastructure ===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Implementations of the classes used to support Liveness Analysis
```

Comment:
Nit: Maybe mention this tracks liveness of all physical registers, including reserved ones.